Extension/ui/settings.html

```diff
@@ -3,129 +3,411 @@
-            <img src="{{root}}/LanguageCCPP_color_128x.png" alt="Microsoft C/C++ Extension" title="Microsoft C/C++ Extension">
-          </td>
-          <td>
-               Microsoft <br>
-               C/C++ Extension
-          </td>
-        </table>
-      </td>
-  </tr>
-
-  <!-- row 1 -->
-  <!-- Save -->
-  <tr>
-      <td>
-          <p>Saves to active configuration <span id=activeConfig></span> in <a href="command:C_Cpp.ConfigurationEditJSON" title="Edit configurations in JSON file">c_cpp_properties.json</a></p>
-      </td>
-  </tr>
-
-  <!-- row 2 -->
-  <!-- compiler path -->
-    <tr>
-      <td>
-        Compiler path
-       <p>
-          <!--<span id="compilerPathInvalid">compiler path is invalid.</span><br>-->
-          <input type="text" id="compilerPath" size="54">
-       </p>
-       <p>
-      </td>
-    </tr>
-
-  <!-- row 4 -->
-  <!-- IntelliSense mode -->
-  <tr>
-      <td>
-          IntelliSense mode
-        <p>
+            <td>
+              <img src="{{root}}/LanguageCCPP_color_128x.png" height="76" width="76" alt="Microsoft C/C++ Extension" title="Microsoft C/C++ Extension" style="padding-right: 10px">
+            </td>
+            <td>
+                <div style="font-size: 20px; font-weight: 400; line-height: 24px">Microsoft<br>C/C++ Extension</div>
+            </td>
+          </table>
+    </div>
+
+    <hr>
+    <div style="height: 30px; display: block"></div>
+
+    <div style="padding-right: 10px;">
+        Command palette to open this editor<br>
+        <div class="codeblock">C/C++: Edit configurations (UI)</div>
+
+        <div style="height: 46px; display: block"></div>
+
+        Command palette to open <a href="command:C_Cpp.ConfigurationEditJSON" title="Edit configurations in JSON file">c_cpp_properties.json</a><br> 
+        <div class="codeblock">C/C++: Edit configurations (JSON)</div>
+    </div>
+
+  </div> <!-- sidebar end -->
+
+  <!-- main -->
+  <div id="main" class="main">
+
+    <!-- title -->
+    <div style=" height: 90px; display: block">
+      <div class="main-title">IntelliSense Configurations</div>
+      <div>
+        Use this editor to edit basic IntelliSense settings for the active configuration.
+        To edit multiple IntelliSense configurations and additional settings go to <a href="command:C_Cpp.ConfigurationEditJSON" title="Edit configurations in JSON file">c_cpp_properties.json</a>.
+      </div>
+    </div>
+
+    <hr>
+    <div style="height: 30px; display: block"></div>
+
+    <!-- sections -->
+    <div class="section">
+      <div class="section-title">Configuration identifier</div>
+      <div class="section-text">
+        The identifier of the active configuration. <code>Mac</code>, <code>Linux</code>, and <code>Win32</code> are special identifiers for configurations that will be auto-selected on those platforms, but the identifier can be anything.
+      </div>
+      <div class="section-input"><input id=activeConfig style="width: 290px"></div>

```

Comment:
Are users allowed to rename the config here?

Reply:
Yes, they can rename the config here the same as they can rename it in .json

Reply:
Changed this field to ready-only and added a link to select a different configuration. If user changes the name, the new name does not show up in the configuration list. Will need to add logic to update the list if there is a rename.